beacon-chain/state/fieldtrie/field_trie_helpers.go

```diff
@@ -65,13 +66,51 @@ func validateElements(field types.FieldIndex, dataType types.DataType, elements
 // fieldConverters converts the corresponding field and the provided elements to the appropriate roots.
 func fieldConverters(field types.FieldIndex, indices []uint64, elements interface{}, convertAll bool) ([][32]byte, error) {
 	switch field {
-	case types.BlockRoots, types.StateRoots, types.RandaoMixes:
-		val, ok := elements.([][]byte)
+	case types.BlockRoots:
+		bVal, ok := elements.([][]byte)
 		if !ok {
-			return nil, errors.Errorf("Wanted type of %v but got %v",
-				reflect.TypeOf([][]byte{}).Name(), reflect.TypeOf(elements).Name())
+			rVal, ok := elements.(*customtypes.BlockRoots)
+			if !ok {
+				return nil, errors.Errorf("Incorrect type used for block roots")
+			}
+			roots := make([][]byte, len(rVal))
+			for i, r := range rVal {
+				tmp := r
+				roots[i] = tmp[:]
+			}
+			return handleByteArrays(roots, indices, convertAll)
```

Comment:
do you think we can have another method here to handle `[][32]byte` instead of only `[][]byte`. If you look at how `handleByteArrays works it always assumes `[][]byte` . We are basically going back to where we started : from fixed sized arrays to slices and then back. 
```go
// handleByteArrays computes and returns byte arrays in a slice of root format.
func handleByteArrays(val [][32]byte, indices []uint64, convertAll bool) ([][32]byte, error) {
	length := len(indices)
	if convertAll {
		length = len(val)
	}
	roots := make([][32]byte, 0, length)
	rootCreator := func(input [32]byte) {
		roots = append(roots, input)
	}
	if convertAll {
		for i := range val {
			rootCreator(val[i])
		}
		return roots, nil
	}
	if len(val) > 0 {
		for _, idx := range indices {
			if idx > uint64(len(val))-1 {
				return nil, fmt.Errorf("index %d greater than number of byte arrays %d", idx, len(val))
			}
			rootCreator(val[idx])
		}
	}
	return roots, nil
}
```

Something like the above but only for `[][32]byte`